### ISSUE TITLE
Durable File Sink stops triggering timer

### ISSUE THREAD

**Issue by Nehmiabm** [FantasticFiasco/serilog-sinks-http#105]
First of all I would like to say what a fantastic library this is. We were using the durable file sink to buffer logs on disk and send to Azure Log Analytics via the HTTP Data Collector API. It was working all great until I started to notice a weird behavior recently where the bookmark stops updating and timer not triggering the HTTP call to send the log events to Azure Log Analytics. What's weird is the timer usually stops triggering at a specific time around 12:01AM. **The file event still continues getting logged to the disk but the bookmark stops getting updated**. If I delete the logs and restart the application, the timer starts to work and bookmark gets updated and continues sending log events to Azure Log Analytics until around 12:01AM where bookmark stops updating and timer stops triggering again.

It's difficult to reproduce the issue locally as this happens when we are logging events continuously throughout the day and timer stops triggering usually around 12:00AM.  Do you have any pointers as to what could cause this? Any help would greatly be appreciated.

Thank you!

**Comment by FantasticFiasco** [comment]
Hi!

This seems interesting, I agree we need more information on the issue. Errors related to Serilog are logged on the [SelfLog](https://github.com/serilog/serilog/wiki/Debugging-and-Diagnostics#selflog). Would it be possible to collect those prior to, and after the issue occurs?

**Comment by Nehmiabm** [comment]
Thank you for getting back to me shortly. I appreciate it!

Even though I have enabled self logging, there is no error in the self log file hence no errors were encountered. Today I observed that for some of the logs that stopped getting picked up, I saw the bookmark got updated but is pointing to an old log file and stuck pointing to that file and the offset updating very slowly. Because of this events are lagging an average of 3 hours behind,  hence there's a big lag on sending events to Log Analytics real time. These logs are enabled for Web Api services which serve moderate number of requests and we are logging each request to send to log analytics. 

Below is the settings I've configured for the sink;

bufferFileSizeLimitBytes : 12000000
retainedBufferFileCountLimit: 10
batchPostingLimit : 30
Period : 300

I didn't expect it to be very slow to read through the files and update the bookmark based on the above configuration. We log maximum of 200MB of data per service in a day with the above configuration. 

Any thoughts?

Thank you!

**Comment by FantasticFiasco** [comment]
This is just off the top of my head, I'll dive down into the details tomorrow, but it seems to me that `batchPostingLimit` (_The maximum number of events to post in a single batch. Default value is 1000_) is quite low, and `period` (_The time to wait between checking for event batches. Default value is 2 seconds._) is quite high.

With this configuration we send a maximum of 30 log events each 5 minutes. That 200MB of data, is it possible to estimate the number of log events that represents? Because with your current configuration, according to my calculations, you can send at most 8640 log events per day. Is that sufficient for the amount of log events you generate?

Is the bandwidth between your web instances and the log server limited, is that the reason your configuration?

**Comment by Nehmiabm** [comment]
Thank you for your fast response again!

I just checked and we log approximately 15,000 log events per single log file. The only reason we set the configurations that way was to reduce the number of round trips to Log Analytics server, thinking that the 5 minutes delay will help us to accumulate the events and send them in a batch in a single request. The reason the batch limit is low was because Azure Log Analytics has a limit on payload size per single request (Maximum allowed payload size per request is 32MB). We are flexible to tune the configurations, what best configuration do you recommend based on this information?

Thank you so much!

**Comment by FantasticFiasco** [comment]
I think you changing the period from 300 seconds to 10 is a great improvement. The duration between a log event being generated and then finally ending up on a log server where your developers can consume them is reduced, thus improving the feedback loop and productivity. But please make sure that your log server can handle the increased number of incoming requests, preferably by monitoring it.

I still think a batch size of 30 is on the lower end. The max payload size of 32MB on Azure Log Analytics is a hard limit that we should respect, but the way the batch size is set you estimate each log event in the batch to be around 1MB. Is that accurate? Do you produce log events of that size? If not I would increase the batch size to reduce the risk falling behind.

**Comment by Nehmiabm** [comment]
I made the adjustments based on our discussion and your recommendations and events are getting processed to Log Analytics with no lag in real time. As you pointed out each log event is not even close to 1MB, it's way less than that, so I adjusted the batch size to 500 and the Period to 10 seconds. It's working as expected and I'm happy to see the events reach Log Analytics instantly.

Thank you for responding quickly and identifying the underlying cause of the slowness. I appreciate all your help, feel free to close this issue :-)
